src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceIndexingOperationListenerImpl.java

```diff
@@ -34,11 +34,13 @@
 public final class ComplianceIndexingOperationListenerImpl extends ComplianceIndexingOperationListener {
 
     private static final Logger log = LogManager.getLogger(ComplianceIndexingOperationListenerImpl.class);
+    private final String indexName;
     private final AuditLog auditlog;
     private volatile IndexService is;
 
-    public ComplianceIndexingOperationListenerImpl(final AuditLog auditlog) {
+    public ComplianceIndexingOperationListenerImpl(final String indexName, final AuditLog auditlog) {
         super();
+        this.indexName = indexName;
```

Comment:
is it necessary to have `indexName`? How it is different from `ShardId.getIndexName()`?

Reply:
Looks like the behavior is going to be the same. 
I wouldn't assume ES will pass a different index shardId for a given registered index module. However will use ShardId.getIndexName() as it is closer to truth of the operation being performed.